### PR TITLE
bug fix - modified inputs

### PR DESCRIPTION
**Pull request by caleb-easterly**
This fixes a pretty nasty bug in which the tool inputs were modified on disk. I'm now copying all inputs to temporary files to be extra safe.

### CODE REVIEW

**Comment by bgruening** [comment]
Please bump the tool version number.
We should also report this upstream. Modifying inputs is a no-go.

**Comment by caleb-easterly** [comment]
Re: version number. I've wondered about this situation before - the version of moff is the same (1.2.1), but the wrapper has changed. Should I call the new version 1.2.1.1 or something like that? 

Yes, this makes me wonder if it's possible to make history items read-only, or something like that.

**Comment by bgruening** [comment]
👍  to 1.2.1.1. You can make dataset inputs readonly, but this needs to be done by the admin of your instance by e.g. using Docker/Singularity ...

File: tools/moFF/moff.xml [hunk 1]
```diff
@@ -421,6 +421,8 @@ It currently uses a Go library to directly extract data from Thermo Raw spectrum
 eliminating the need for conversions from other formats.
 Moreover, moFF also allows one to work directly with mzML files.
 
+This is moFF version @VERSION@.
```
Comment (bgruening) [inline tools/moFF/moff.xml]: Don't do it like this. Use https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-version-command

File: tools/moFF/moff.xml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<tool id="proteomics_moff" name="moFF" version="@VERSION@">
+<tool id="proteomics_moff" name="moFF" version="1.2.1.1">
```
Comment (bgruening) [inline tools/moFF/moff.xml]: here you can keep `@VERSION@.1`

Comment (caleb-easterly) [reply tools/moFF/moff.xml]: ah, good point.

File: tools/moFF/moff.xml (hunk 1 above)
Comment (caleb-easterly) [reply tools/moFF/moff.xml]: I thought about this, but I don't think moff has a `moff.py --version`, as far as I can tell. Is this way okay or is there a conda workaround?

Comment (bgruening) [reply tools/moFF/moff.xml]: No conda workaround. But you could echo "@VERSION@". Which is also bad but better then nothing.
moff should implement the version ... :)

**Pull request merged**